Fake content for oai provider

kazooo opened the pull request:
Tento PR zachytává chyby, které jsou způsobeny v důsledku dat (https://github.com/moravianlibrary/kramerius/issues/207). Chyba se zachytí a její message je vypsána přímo do metadat. Je to lepší než aby kolaboval celý proces sklízení.

kazooo committed:
Fake content for oai provider

kazooo committed:
Issue #565: fix oai export (missing Fedora object)

leiblix commented on search/src/java/cz/incad/Kramerius/oai/OaiServlet.java:
```diff
@@ -109,16 +109,31 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
 
     }
+    
+    private void sendFakeContent(HttpServletResponse resp) {
```
přejmenoval bych na sendErrorMessage

MartinRumanek replied:
Nedával bych tam code="404", protože ty chyby se můžou lišit....lepší bude mít atribut message a před tam message z IOException. Pokud IOEXception nenese rozumnou hlášku bude pravděpodobně nutné upravit DrKrameriusV4Writer.

leiblix commented on search/src/java/cz/incad/Kramerius/oai/OaiServlet.java:
```diff
@@ -109,16 +109,31 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
+        resp.setCharacterEncoding("UTF-8");
+        resp.setContentLength(emptyXml.length());
+
+        try {
```
Zavírat zdroje by se mělo vždy v bloku finally, nebo nejlépe v tomto případě použít try-with-resources (už můžeme konečně používat konstruky z novější Javy - v IDE si to přepni rovnou na Javu 8)

MartinRumanek commented on search/src/java/cz/incad/Kramerius/oai/OaiServlet.java:
```diff
@@ -109,16 +109,31 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
+        try {
+            resp.getWriter().write(emptyXml);
+            resp.getWriter().flush();
+            resp.getWriter().close();
```
Close - zavírat zdroje bys měl vždy v bloku finally. Kramerius už konečně opustil JDK 6, takže můžeme používat novější konstrukty. Např. try-with-resources, PrintWriter implementuje rozhraní closeable, takže by to mělo jít.

leiblix commented on search/src/java/cz/incad/Kramerius/oai/OaiServlet.java:
```diff
@@ -109,16 +109,31 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
     }
+    
+    private void sendFakeContent(HttpServletResponse resp) {
+        String emptyXml = "<error code=\"404\">";
```
Neposílat 404, ale spíše getMessage z IOException. Pokud neobsahuje dobrou message bude nutné upravit DrKrameriusV4Writer.java

jkremlacek replied:
Proměnnou bych potom přejmenoval na `xmlContent`, `emptyXml` by mohlo evokovat, že se jedná o prázdný řetězec.

leiblix reviewed (changes requested):
Jo, to by mohlo fungovat. Testoval jsi to? Nebo to zkusíme rovnou na produkci?

Před začleněním do codebase budu chtít vyřešit pár drobností, které jsem napsal do komentářů.

jkremlacek commented on search/src/java/cz/incad/Kramerius/oai/OaiServlet.java:
```diff
@@ -96,7 +96,7 @@ public void init() throws ServletException {
     }
 
-    @Override
+   @Override
```
Chyba odsazení.

kazooo committed:
Fix corrections

Nevim jenom jak otestovat ty opravy. K tomu bych potreboval udaje monografie v solru, ktera neni ve fedore.

leiblix commented:
Prosím teda prosím otestovat na produkci...s nasazením ti může pomoci @dsecik .

kazooo committed:
last changes

leiblix commented:
Hezké 👍 

Prosím udělej squash přes ty tři commity git rebase -i HEAD~3 (http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html).

A commit message přepiš na:

Issue #565: fix oai export (missing Fedora object)

_______
Pak už to zamerguju do upstreamu. Díky!

kazooo committed:
Merge branch 'oai-fix-error' of https://github.com/xermak00/kramerius into oai-fix-error
